WCFDataService/Client/System/Data/Services/Client/Materialization/HttpWebResponseMessage.cs

```diff
@@ -98,6 +100,21 @@ internal HttpWebResponseMessage(HeaderCollection headers, int statusCode, Func<S
         }
 
         /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="headers">The headers.</param>
+        /// <param name="statusCode">The status code.</param>
+        /// <param name="getResponseStream">A function returning the response stream.</param>
+        /// <param name="underlyingResponseMessage">The underlying response message that should be disposed together with this instance.</param>
+        internal HttpWebResponseMessage(HeaderCollection headers, int statusCode, Func<Stream> getResponseStream, IODataResponseMessage underlyingResponseMessage)
```

Comment:
I tried this already. A debug build will fail on this assert in HttpWebResponse.GetStream():
            Debug.Assert(!this.streamReturned, "The GetStream can only be called once.");
As there are actually two HttpWebResponseMessages, the first one is constructed from  HttpWebResponseMessage(HttpWebResponse httpResponse), the other one constructed from HttpWebResponseMessage(HeaderCollection headers, int statusCode, Func<Stream> getResponseStream, IODataResponseMessage underlyingResponseMessage) that actually wraps the first one. This would lead to GetStream being called twice on the first HttpWebResponseMessage.